Dialog/TilesetEditDialog.cpp

```diff
@@ -0,0 +1,941 @@
+#include "TilesetEditDialog.h"
+#include "ui_TilesetEditDialog.h"
+
+#include <QFile>
+#include <QFileDialog>
+#include <QInputDialog>
+#include <QMessageBox>
+
+TilesetEditDialog::TilesetEditDialog(QWidget *parent, DialogParams::TilesetEditParams *tilesetEditParam) :
```

Comment:
Why is this a pointer? Perhaps it would be easier to manage the TilesetEditParams struct if it is not a pointer. Otherwise I worry that it may be easy to forget to de-allocate.

Reply:
but there is a Tileset* inside TilesetEditParams, so it had better to be a pointer too